app/models/concerns/permalinkable.rb

```diff
@@ -9,7 +9,15 @@ module Permalinkable
 
   module ClassMethods
     def find_by_param!(param)
-      find_by_permalink!(param)
+      begin
+        find_by_permalink!(param)
+      rescue ActiveRecord::RecordNotFound
+        if param =~ /^\d+$/
```

Comment:
I believe it's valid for a Project to have a name (and thus a permalink) of "12345".  If you branch at the top of the method, you'll never be able to find a project with that name.  The `find_by_id!` is a fallback so that we can create urls of the form `/project/{id}/stages` as well as `/project/{permalink}/stages`.  I can't think of a way of doing that without rescuing an Exception.